src/change/sdp.rs

```diff
@@ -993,24 +1007,63 @@ fn update_media(
     // SSRC changes
     // This will always be for ReceiverSource since any incoming a=ssrc line will be
     // about the remote side's SSRC.
-    let infos = m.ssrc_info();
+    let infos = m.ssrc_info(false);
 
     let main = infos.iter().filter(|i| i.repairs.is_none());
+    let mid = media.mid();
+
+    fn log_change(mut log_message: String, repair_ssrc: Option<Ssrc>, mid: Mid, rid: Option<Rid>) {
+        if let Some(rtx_ssrc) = repair_ssrc {
+            write!(log_message, ", RTX: SSRC: {}", rtx_ssrc).expect("write! to string never fails");
+        }
+        write!(log_message, " for MID: {}", mid).expect("write! to String never fails");
 
-    if m.simulcast().is_none() {
-        // Only use pre-communicated SSRC if we are running without simulcast.
-        // We found a bug in FF where the order of the simulcast lines does not
-        // correspond to the order of the simulcast declarations. In this case
-        // it's better to fall back on mid/rid dynamic mapping.
+        if let Some(rid) = rid {
+            write!(log_message, " and RID: {}", rid).expect("write! to String never fails");
+        }
+        info!("{}", log_message);
+    }
 
-        for i in main {
-            // TODO: If the remote is communicating _BOTH_ rid and a=ssrc this will fail.
-            info!("Adding pre-communicated SSRC: {:?}", i);
-            let repair_ssrc = infos
-                .iter()
-                .find(|r| r.repairs == Some(i.ssrc))
-                .map(|r| r.ssrc);
-            streams.expect_stream_rx(i.ssrc, repair_ssrc, media.mid(), None);
+    // Use pre-communicated SSRCs if available and prefer them over dynamically signalled SSRCs via
+    // rid and repaired-rid. There's a bug in Firefox that causes it to emit incorrect mappings via
+    // repaired-rid while the values signalled in SDP are correct.
+    for i in main {
+        // If the remote is communicating _BOTH_ rid/repaired-rid and a=ssrc/a=ssrc-group we will
+        // adopt the mapping from the SDP as described above. Since we have already established the
+        // mapping any further mapping via rid/repaired-rid headers for these SSRCs will be ignored
+        let repair_ssrc = infos
+            .iter()
+            .find(|r| r.repairs == Some(i.ssrc))
+            .map(|r| r.ssrc);
+        let rid = i.rid;
+
+        if let Some(existing_stream) = streams.stream_rx_by_mid_rid(mid, rid) {
+            // The remote changed its mapping, this happens in Firefox when transitioning an m-line
```

Comment:
Should check if the SSRC of either the main RPT stream or the repair stream actually changes here